### PR TITLE
(feat) enhancement on the LIPANA MPESA STK push and  initial claims

### PR DESCRIPTION
**Pull request by its-kios09**
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
- (feat) added error handling by add inline notification 

## Screenshots
https://github.com/palladiumkenya/kenyaemr-esm-3.x/assets/67967749/14209079-456e-4a44-adb4-253a903c46e9

*None.*

## Related Issue

*None.*

## Other
- To run the LIPA NA MPESA you need credentials from daraja API which included the following
**consumer_Key
consumer_Sercet
Passkey
Authorization_URL
Initiate_URL**

- To run it on local instance 
`google-chrome --user-data-dir=/tmp --disable-web-security
`

### CODE REVIEW

**Comment by makombe** [comment]
Thanks @its-kios09. Something that has just come to mind can we ensure that `Initiate payment` is only shown where payment method is MPESA. This can happen in a follow up PR not necessary on this. Thanks

**Comment by ojwanganto** [comment]
Thanks, @its-kios09. Should we rename the "Initiate payment" button to read something processing payment?

**Comment by its-kios09** [comment]
@makombe @ojwanganto  I was thinking the payment method is we use it for reconciliation of payments due to the ref number textfield or what do you think?
On the renaming, maybe we should look for a simple name

**Comment by ojwanganto** [comment]
You are right, @its-kios09.

**Comment by its-kios09** [comment]
Let me work on the suggestion

**Comment by ojwanganto** [comment]
@its-kios09 you should not close the PR. I believe you can still just update the PR with additional changes.

File: packages/esm-billing-app/src/m-pesa/mpesa-resource.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ const swrOption = {
   errorRetryCount: 2,
 };
 
-export const usePaymentModes = (excludeWaiver: boolean = true) => {
+export const usePaymentModes = () => {
```
Comment (ojwanganto) [inline packages/esm-billing-app/src/m-pesa/mpesa-resource.tsx:17]: Should this file be named something else other than mpesa? I can see it has other payment modes other than mpesa.

File: packages/esm-billing-app/src/invoice/invoice.component.tsx [hunk 2]
```diff
@@ -14,7 +14,7 @@ import Payments from './payments/payments.component';
 import PrintReceipt from './printable-invoice/print-receipt.component';
 import PrintableInvoice from './printable-invoice/printable-invoice.component';
 import styles from './invoice.scss';
-import MakeClaims from './claims/make-claims.component';
+import MakeClaims from '../claims/make-claims.component';
```
Comment (ojwanganto) [inline packages/esm-billing-app/src/invoice/invoice.component.tsx:17]: Has this been used in here?

Comment (its-kios09) [reply packages/esm-billing-app/src/invoice/invoice.component.tsx:17]: Nop, I Have decided to combine mpesa integration and claims functionality on the same branch as Donald suggestion

File: packages/esm-billing-app/src/claims/claims-dashboard/claims-header/claims-header.component.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+interface ClaimsHeaderProps {
+  patient: fhir.Patient;
+  bill: MappedBill;
+}
+
+const ClaimsHeader: React.FC<ClaimsHeaderProps> = ({ patient, bill }) => {
+  const { t } = useTranslation();
+  console.log('patient', patient);
+  console.log('billUuid', bill);
```
Comment (ojwanganto) [inline packages/esm-billing-app/src/claims/claims-dashboard/claims-header/claims-header.component.tsx:18]: you will not need these

File: packages/esm-billing-app/src/claims/claims-dashboard/claims-dashboard.component.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  const { bill, isLoading: isLoadingBill, error } = useBill(billUuid);
+
+  if (isLoadingPatient && isLoadingBill) {
+    return (
+      <div className={styles.invoiceContainer}>
+        <InlineLoading
+          className={styles.loader}
+          status="active"
+          iconDescription="Loading"
+          description="Loading patient header..."
```
Comment (ojwanganto) [inline packages/esm-billing-app/src/claims/claims-dashboard/claims-dashboard.component.tsx:24]: Should we translate this?

File: packages/esm-billing-app/src/invoice/invoice.component.tsx (hunk 2 above)
Comment (ojwanganto) [reply packages/esm-billing-app/src/invoice/invoice.component.tsx:17]: OK

File: packages/esm-billing-app/translations/en.json [hunk 5]
```diff
@@ -44,6 +44,7 @@
   "enterConcept": "Associated concept",
   "enterReferenceNumber": "Enter ref. number",
   "enterSHANumber": "Enter SHA Number",
+  "Error": "",
```
Comment (ojwanganto) [inline packages/esm-billing-app/translations/en.json:47]: What does this do?

File: packages/esm-billing-app/src/claims/claims-dashboard/claims-header/claims-header.component.tsx (hunk 3 above)
Comment (its-kios09) [reply packages/esm-billing-app/src/claims/claims-dashboard/claims-header/claims-header.component.tsx:18]: I have remove them on my next commit

File: packages/esm-billing-app/src/claims/claims-dashboard/claims-dashboard.component.tsx (hunk 4 above)
Comment (its-kios09) [reply packages/esm-billing-app/src/claims/claims-dashboard/claims-dashboard.component.tsx:24]: I will be handling translation today

File: packages/esm-billing-app/translations/en.json (hunk 5 above)
Comment (its-kios09) [reply packages/esm-billing-app/translations/en.json:47]: Let me look at it

File: package.json [hunk 6]
```diff
@@ -30,7 +30,7 @@
     "zod": "^3.21.4"
   },
   "devDependencies": {
-    "@openmrs/esm-framework": "next",
+    "@openmrs/esm-framework": "^5.6.1-pre.1895",
```
Comment (donaldkibet) [inline package.json:33]:
```suggestion
    "@openmrs/esm-framework": "next",
```
we only need to update the `yarn.lock` file for such updates

File: packages/esm-billing-app/src/claims/claims-dashboard/claims-dashboard.component.tsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+      <div className={styles.errorContainer}>
+        <ErrorState headerTitle={t('createClaimError', 'Create Claim error')} error={error} />
+      </div>
+    );
+  }
+
+  return (
+    <>
+      <ClaimsHeader patient={patient} bill={bill} />
+    </>
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/claims/claims-dashboard/claims-dashboard.component.tsx:41]:
```suggestion    
      <ClaimsHeader patient={patient} bill={bill} />    
```

File: packages/esm-billing-app/src/claims/claims-wrap/claims-main-component.tsx [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+import { computeTotalPrice } from '../../utils';
+import { ClaimsBreakDown } from '../claims-dashboard/claims-breakdown/claims-breakdown.component';
+import { convertToCurrency } from '../../helpers';
+import { useTranslation } from 'react-i18next';
+
+interface ClaimsMainHeaderProps {
+  bill: MappedBill;
+}
+
+const ClaimMainComponent: React.FC<ClaimsMainHeaderProps> = ({ bill }) => {
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/claims/claims-wrap/claims-main-component.tsx:16]:
```suggestion
const ClaimMainComponent: React.FC<ClaimsMainProps> = ({ bill }) => {
```

**Review by donaldkibet — CHANGES_REQUESTED** [review]

File: package.json (hunk 6 above)
Comment (its-kios09) [reply package.json:33]: I have issue with navigate i decided to use` yarn up openmrs @openmrs/esm-framework` and it worked

File: packages/esm-billing-app/src/claims/claims-dashboard/claims-dashboard.component.tsx (hunk 7 above)
Comment (its-kios09) [reply packages/esm-billing-app/src/claims/claims-dashboard/claims-dashboard.component.tsx:41]: sorted

File: packages/esm-billing-app/src/claims/claims-wrap/claims-main-component.tsx (hunk 8 above)
Comment (its-kios09) [reply packages/esm-billing-app/src/claims/claims-wrap/claims-main-component.tsx:16]: sorted

**Review by donaldkibet — APPROVED** [review]

**Comment by donaldkibet** [comment]
Thanks @its-kios09, merging this in

**Pull request merged**